core/ledger/confighistory/mgr_test.go

```diff
@@ -300,100 +296,210 @@ func newTestEnvForSnapshot(t *testing.T) *testEnvForSnapshot {
+			{&compositeKey{ns: "lscc", key: "chaincode3~collection", blockNum: 50}, cc3configBytes},
+		}
+
+		return storedKVs, ccConfigInfo
 	}
-	verifyExportedConfigHistory(t, env.testSnapshotDir, fileHashes, expectedCollectionConfigs)
+
+	cleanup := func() {
+		require.NoError(t, os.RemoveAll(filepath.Join(env.testSnapshotDir, snapshotDataFileName)))
+		require.NoError(t, os.RemoveAll(filepath.Join(env.testSnapshotDir, snapshotMetadataFileName)))
+	}
+
+	t.Run("confighistory is empty", func(t *testing.T) {
+		retriever := env.mgr.GetRetriever("ledger1", nil)
+		fileHashes, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+		require.Empty(t, fileHashes)
+		files, err := ioutil.ReadDir(env.testSnapshotDir)
+		require.NoError(t, err)
+		require.Len(t, files, 0)
+	})
+
+	t.Run("export confighistory", func(t *testing.T) {
+		// setup ledger1 => export ledger1
+		storedKVs, _ := setup("ledger1")
+		retriever := env.mgr.GetRetriever("ledger1", nil)
+		fileHashes, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+		verifyExportedConfigHistory(t, env.testSnapshotDir, fileHashes, storedKVs)
+		cleanup()
+	})
+
+	t.Run("import confighistory and verify queries", func(t *testing.T) {
+		// setup ledger2 => export ledger2 => import into ledger3
+		_, ccConfigInfo := setup("ledger2")
+		retriever := env.mgr.GetRetriever("ledger2", nil)
+		_, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+
+		importConfigsBatchSize = 100
+		require.NoError(t, env.mgr.ImportConfigHistory("ledger3", env.testSnapshotDir))
+		dummyLedgerInfoRetriever := &dummyLedgerInfoRetriever{
+			info: &common.BlockchainInfo{Height: 1000},
+			qe:   &mock.QueryExecutor{},
+		}
+
+		retriever = env.mgr.GetRetriever("ledger3", dummyLedgerInfoRetriever)
+		verifyImportedConfigHistory(t, retriever, ccConfigInfo)
+		cleanup()
+	})
+
+	t.Run("export from an imported confighistory", func(t *testing.T) {
+		// setup ledger4 => export ledger4 => import into ledger5 => export ledger5
+		storedKVs, _ := setup("ledger4")
+		retriever := env.mgr.GetRetriever("ledger4", nil)
+		_, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+
+		importConfigsBatchSize = 100
+		require.NoError(t, env.mgr.ImportConfigHistory("ledger5", env.testSnapshotDir))
+		cleanup()
+
+		retriever = env.mgr.GetRetriever("ledger5", nil)
+		fileHashes, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+		verifyExportedConfigHistory(t, env.testSnapshotDir, fileHashes, storedKVs)
+		cleanup()
+	})
+
+	t.Run("import confighistory error cases", func(t *testing.T) {
+		setup("ledger6")
+		retriever := env.mgr.GetRetriever("ledger6", nil)
+		_, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+
```

Comment:
Not sure what is the significance of these lines in this test

Reply:
I don't understand. I am exporting ledger 6 so that I don't need to create some random snapshot file which would otherwise give me `error reading snapshot file.... EOF`

Reply:
Understood. I found it confusing because of following two reasons
- This was not needed for the immidiate following test.
- As you follow through the test code, one has to keep track of how the content in the file is getting changed. 

I think that these error tests deserve to be separate subtests but I leave that to you. Otherwise, if you can produce the error causing condition explicitly (like you did at line 485) each time a fresh, that may be more consumable. In a related [test](https://github.com/hyperledger/fabric/blob/327765d59a9dd095e0282eda18c8e6306594daf1/common/ledger/blkstorage/snapshot_test.go#L342), I found this pattern simpler to maintain. But as I said, it's not a blocker for me and leave to you.

Reply:
I understood your confusion. When I add `Remove()` function, I will refactor this sub-test too.